h2o-algos/src/main/java/hex/gam/GAM.java

```diff
@@ -212,61 +233,36 @@ private void validateGamParameters() {
       error("_gam_columns", "must specify columns names to apply GAM to.  If you don't have any," +
               " use GLM.");
-    else {  // check and make sure gam_columns column types are legal
-      Frame dataset = _parms.train();
-      List<String> cNames = Arrays.asList(dataset.names());
-      for (int index = 0; index < _parms._gam_columns.length; index++) {
-        String cname = _parms._gam_columns[index];
-        if (!cNames.contains(cname))
-          error("gam_columns", "column name: " + cname + " does not exist in your dataset.");
-        if (dataset.vec(cname).isCategorical())
-          error("gam_columns", "column " + cname + " is categorical and cannot be used as a gam " +
-                  "column.");
-        if (dataset.vec(cname).isBad() || dataset.vec(cname).isTime() || dataset.vec(cname).isUUID() ||
-                dataset.vec(cname).isConst())
-          error("gam_columns", String.format("Column '%s' of type '%s' cannot be used as GAM column. Column types " +
-                  "BAD, TIME, CONSTANT and UUID cannot be used.", cname, dataset.vec(cname).get_type_str()));
-        if (!dataset.vec(cname).isNumeric())
-          error("gam_columns", "column " + cname + " is not numerical and cannot be used as a gam" +
-                  " column.");
-      }
-    }
-    if ((_parms._bs != null) && (_parms._gam_columns.length != _parms._bs.length))  // check length
-      error("gam colum number", "Number of gam columns implied from _bs and _gam_columns do not match.");
-    if (_parms._bs == null) // default to bs type 0
-      _parms._bs = new int[_parms._gam_columns.length];
-    if (_parms._num_knots == null) {  // user did not specify any knot numbers, we will use default 10
-      _parms._num_knots = new int[_parms._gam_columns.length];  // different columns may have different
-      for (int index = 0; index < _parms._gam_columns.length; index++) {  // for zero value _num_knots, set to valid number
-        if (_parms._num_knots[index] == 0) {
-          long numRowMinusNACnt = _train.numRows() - _parms.train().vec(_parms._gam_columns[index]).naCnt();
-          _parms._num_knots[index] = numRowMinusNACnt < 10 ? (int) numRowMinusNACnt : 10;
-        }
-      }
-    }
-    int cindex = 0;
-    for (int numKnots : _parms._num_knots) {  // check to make sure numKnot is valid
-      long eligibleRows = _train.numRows() - _parms.train().vec(_parms._gam_columns[cindex]).naCnt();
-      if (numKnots > eligibleRows) {
-        error("_num_knots", " number of knots specified in _num_knots: " + _parms._num_knots[cindex] + " exceed number " +
-                "of rows in training frame minus NA rows: " + eligibleRows + ".  Reduce _num_knots.");
-      }
-      cindex++;
+    } else { // check and make sure gam_columns column types are legal
+      if (_parms._bs == null)
+        setDefaultBSType(_parms);
+      if ((_parms._bs != null) && (_parms._gam_columns.length != _parms._bs.length))  // check length
```

Comment:
I personally prefer using curly braces even for one line if statements and rather avoid using redundant braces inside if conditions but leaving this up to you

Reply:
Got it.  The reason for the parenthesis inside if is because I ran into the problem of not remembering the priority of the various operators.  That is why to be sure, I just add parenthesis.